ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Byron.hs

```diff
@@ -515,61 +378,214 @@ instance Condense (Header (ByronBlock cfg)) where
         . unByronHeader
         $ hdr
 
-instance Condense (ChainHash (ByronBlock cfg)) where
+{-------------------------------------------------------------------------------
+  Epoch Boundary Blocks
+-------------------------------------------------------------------------------}
+
+newtype ByronBlockOrEBB cfg = ByronBlockOrEBB
+  { unByronBlockOrEBB :: CC.Block.ABlockOrBoundary ByteString
+  } deriving (Eq, Show)
```

Comment:
For byron-proxy, I'll need a type which includes Byron blocks and EBBs, but which also can be serialised. Since there's no way to serialise an `ABlockOrBoundary ByteString`, what I've done there is use

```Haskell
type Block = Annotated (ABlockOrBoundary ByteString) ByteString
```

as the block type, and encoded it using CBOR tag 24 so that when I decode it from the wire I can give the top-level annotation (I have the bytes).

Now I'm working on using `ChainDB` in byron-proxy, so I'll face the same problem with the block type chosen to go in the `ChainDB`. I'll need the top-level annotation there as well. Perhaps we could change `ByronBlockOrEBB` to include this?

Reply:
This is also important for using the `ChainDB` to serve a `Byron` peer. That requires coming up with legacy `cardano-sl` `Block` and `BlockHeader` values, but if I'm using a `ChainDB` that uses new `cardano-ledger` variants, I can't do that without having the bytes from which they were decoded.

Reply:
I don't understand this comment - why can't you serialise an `ABlockOrBoundary ByteString`? It's already annotated, and we have explicit serialisers for it (`toCBORABOBBlock` and `toCBORABOBBoundary`, but we could easily wrap these together).

Reply:
`ABlockOrBoundary ByteString` couldn't be serialized, but now can since you recently merged a pull request into `cardano-ledger`.

Reply:
For `BoundaryValidationData a` the annotation `a` is for the header bytes only, so `ABOBBoundary (BoundaryValidationData ByteString)` doesn't have the bytes for the whole block. That's why I put the annotation out front. Looks like the `toCBORABOBBoundary` just makes up a body... I'm not sure if that's actually ok. The Byron peers might not accept it.